v0/src/jsonLogic.js

```diff
@@ -404,26 +404,40 @@ function validateInlineRules(jsonSchema, sampleEmptyObject) {
  * @param {Function} errorMessage - Function to generate custom error message.
  *                                  Receives the invalid rule part and should throw an error message string.
  */
+
 function checkRuleIntegrity(
   rule,
   id,
   data,
   errorMessage = (item) =>
-    `[json-schema-form] json-logic error: rule "${id}" has no variable "${item.var}".`
+    `[json-schema-form] json-logic error: rule "${id}" has no variable "${item.var}".`,
+  inReduceOrMap = false
 ) {
   Object.entries(rule ?? {}).map(([operator, subRule]) => {
     if (!Array.isArray(subRule) && subRule !== null && subRule !== undefined) return;
     throwIfUnknownOperator(operator, subRule, id);
 
+    // If we are within a reduce or map, we allow references to `accumulator` and/or `current`.
+    // We augment the data so that we still validate any other field.
+    const isReduceOrMap = inReduceOrMap || operator === 'reduce' || operator === 'map';
+
+    const validationData = isReduceOrMap
+      ? {
+          ...data,
+          accumulator: 0,
```

Comment:
Setting the initial value of the accumulator to 0, we're assuming that the final result of the reduce will always be a number. Is this intentional? I'm thinking about a different scenario where you might want to concatenate strings:

```
{"reduce":[
    {"var":"integers"},
    {"+":[{"var":"current"}, {"var":"accumulator"}]},
    ""
]}
```
When running `jsonLogic.apply`, this would result in `0Hello World`, right?

Reply:
mmh, yes, that could be a problem, the logic would work since this is only for checking the integrity of the rules, and when actually running it for your form it will use the right parameters, but it would be a problem if you say have a division instead, let me create a test to see, good point

Reply:
Makes sense, thanks for adding the test 👍